packages/flutter/lib/src/rendering/animated_size.dart

```diff
@@ -0,0 +1,138 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/animation.dart';
+import 'package:meta/meta.dart';
+
+import 'box.dart';
+import 'object.dart';
+import 'shifted_box.dart';
+
+/// A render object that animates its size to its child's size over a given
+/// [duration] and with a given [curve]. In case the child changes its size in
```

Comment:
I think the second sentence could be clearer. I think what you mean is: If the child's size changes during the animation, the animation stops and the render object's size is immediately updated to match the child's size. 

Which begs a question. Why not just reconfigure the animation to target the new child's size?